src/theory/booleans/theory_bool_type_rules.cpp

```diff
@@ -45,6 +49,10 @@ TypeNode BooleanTypeRule::computeType(NodeManager* nodeManager,
   return booleanType;
 }
 
+TypeNode IteTypeRule::preComputeType(NodeManager* nm, TNode n)
```

Comment:
This type rule surprised me because I thought ITE was actually a bultin operator, not a Boolean one, since it's not necessarily Boolean.

Reply:
It belongs to THEORY_BOOL, not exactly sure why, although it can be of any type. Not sure why this the case or if it makes a difference to move ITE to THEORY_BUILTIN.